pkg/build/generic/docker.go

```diff
@@ -0,0 +1,100 @@
+package generic
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/build/common"
+	"github.com/testground/testground/pkg/docker"
+	"github.com/testground/testground/pkg/rpc"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+)
+
+const (
+	buildNetworkName = "testground-build"
+)
+
+var (
+	_ api.Builder = &DockerGenericBuilder{}
+)
+
+type DockerGenericBuilder struct {
+	Enabled bool
+}
+
+type DockerGenericBuilderConfig struct {
+	BuildArgs    map[string]*string `toml:"build_args"`
+	PushRegistry bool               `toml:"push_registry"`
+	RegistryType string             `toml:"registry_type"`
+}
+
+// Build builds a testplan written in Go and outputs a Docker container.
+func (b *DockerGenericBuilder) Build(ctx context.Context, in *api.BuildInput, ow *rpc.OutputWriter) (*api.BuildOutput, error) {
+	cfg, ok := in.BuildConfig.(*DockerGenericBuilderConfig)
+	if !ok {
+		return nil, fmt.Errorf("expected configuration type DockerGenericBuilderConfig, was: %T", in.BuildConfig)
+	}
+
+	cliopts := []client.Opt{client.FromEnv, client.WithAPIVersionNegotiation()}
+
+	var (
+		id       = in.BuildID
+		plansrc  = in.TestPlanSrcPath
+		cli, err = client.NewClientWithOpts(cliopts...)
+	)
+
+	ow = ow.With("build_id", id)
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+
+	opts := types.ImageBuildOptions{
+		Tags:        []string{id, in.BuildID},
+		BuildArgs:   cfg.BuildArgs,
```

Comment:
A comment for our future selves:

```suggestion
		BuildArgs:   cfg.BuildArgs,    // ok if nil
```